### ISSUE TITLE
[Bug]: search error: No Available Query node result

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#12148]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20211119-8a9fb9b
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus-2.0.0rc9.dev11
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others: queryNode.replicas: 3
          dataCoord.enableCompaction: true
```

### Current Behavior

1. create binary collection with shard_num=1
2. insert 3000 entities and flush
3. insert 3000 entities and flush
4. create index `default_binary_index = {"index_type": "BIN_IVF_FLAT", "params": {"nlist": 128}, "metric_type": "JACCARD"}`
5. compact and complete
6. load and search
Return exception:
```
[2021-11-19 20:54:51,505 - DEBUG - ci_test]: (api_response) : {"name": "compact_0JiM78zx", "schema": {"auto_id": false, "description": "", "fields": [{"name": "int64", "description": "", "type": 5, "is_primary": true, "auto_id": false}, {"name": "float", "description": "", "type": 10}, {"name": "binary_vector", "description": "", "type": 100, "params": {"dim":......  (api_request.py:21)
[2021-11-19 20:54:51,780 - DEBUG - ci_test]: (api_request)  : [Collection.insert] args: [      int64   float                                      binary_vector
0         0     0.0  b'\xdaQO\xa2w\xfb\xcb\xa0q\xdf\xfa\xd0\xcf\x19...
1         1     1.0  b'\x1d\xebB\x8e\xd1\xd0\x1b\x023\xfa\xd4\x91\x...
2         2     2.0  b'\xc9\xfd\t%\x95\xb2\r\x08x\xda\x18\x97\xb7\x...
3         3    ......, kwargs: {'timeout': 20} (api_request.py:44)
[2021-11-19 20:54:52,303 - DEBUG - ci_test]: (api_response) : (insert count: 3000, delete count: 0, upsert count: 0, timestamp: 429215315977043969)  (api_request.py:21)
[2021-11-19 20:54:55,477 - DEBUG - ci_test]: (api_request)  : [Collection.insert] args: [      int64   float                                      binary_vector
0         0     0.0  b'\xdaQO\xa2w\xfb\xcb\xa0q\xdf\xfa\xd0\xcf\x19...
1         1     1.0  b'\x1d\xebB\x8e\xd1\xd0\x1b\x023\xfa\xd4\x91\x...
2         2     2.0  b'\xc9\xfd\t%\x95\xb2\r\x08x\xda\x18\x97\xb7\x...
3         3    ......, kwargs: {'timeout': 20} (api_request.py:44)
[2021-11-19 20:54:55,895 - DEBUG - ci_test]: (api_response) : (insert count: 3000, delete count: 0, upsert count: 0, timestamp: 429215316920762369)  (api_request.py:21)
[2021-11-19 20:54:59,010 - DEBUG - ci_test]: (api_request)  : [Collection.create_index] args: ['binary_vector', {'index_type': 'BIN_IVF_FLAT', 'params': {'nlist': 128}, 'metric_type': 'JACCARD'}], kwargs: {'timeout': 20} (api_request.py:44)
[2021-11-19 20:55:03,223 - DEBUG - ci_test]: (api_response) : Status(code=0, message='')  (api_request.py:21)
[2021-11-19 20:55:03,223 - DEBUG - ci_test]: (api_request)  : [Collection.index] args: [], kwargs: {} (api_request.py:44)
[2021-11-19 20:55:03,233 - DEBUG - ci_test]: (api_response) : <pymilvus.orm.index.Index object at 0x7fcb847f8520>  (api_request.py:21)
[2021-11-19 20:55:03,233 - DEBUG - ci_test]: (<pymilvus.orm.index.Index object at 0x7fcb847f8520>, True) (test_compaction.py:321)
[2021-11-19 20:55:03,233 - DEBUG - ci_test]: (api_request)  : [Collection.compact] args: [20], kwargs: {} (api_request.py:44)
[2021-11-19 20:55:03,252 - DEBUG - ci_test]: (api_response) : None  (api_request.py:21)
[2021-11-19 20:55:03,761 - DEBUG - ci_test]: True (collection_wrapper.py:263)
[2021-11-19 20:55:03,762 - DEBUG - ci_test]: (api_request)  : [Collection.get_compaction_plans] args: [20], kwargs: {} (api_request.py:44)
[2021-11-19 20:55:03,771 - DEBUG - ci_test]: (api_response) : 
Compaction Plans:
 - compaction id: 429215318860627969
 - state: 2
 - plans: [
Plan:
 - sources: [429215315990151169, 429215316933869569]
 - target: 429215318860627971
]
   (api_request.py:21)
[2021-11-19 20:55:03,772 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None], kwargs: {'timeout': 20} (api_request.py:44)
[2021-11-19 20:55:04,481 - DEBUG - ci_test]: (api_response) : None  (api_request.py:21)
[2021-11-19 20:55:04,483 - DEBUG - ci_test]: (api_request)  : [Collection.search] args: [[b'\x19x\xc6sY\xc1Z\xa3)|\xdd6\xc6iK{', b';\xe6#7\xda\xa0\x86Z\xe6m\x90S6\x9a8\xab'], 'binary_vector', {'metric_type': 'L2', 'params': {'nprobe': 10}}, 10, None, None, None, 20, -1], kwargs: {} (api_request.py:44)
[2021-11-19 20:55:04,731 - ERROR - ci_test]: Traceback (most recent call last):
  File "/home/zong/zong/projects/milvus/tests/python_client/utils/api_request.py", line 18, in inner_wrapper
    res = func(*args, **kwargs)
  File "/home/zong/zong/projects/milvus/tests/python_client/utils/api_request.py", line 45, in api_request
    return func(*arg, **kwargs)
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 682, in search
    res = conn.search(self._name, data, anns_field, param, limit, expr,
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/stub.py", line 58, in handler
    raise e
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/stub.py", line 42, in handler
    return func(self, *args, **kwargs)
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/stub.py", line 1046, in search
    return handler.search(collection_name, data, anns_field, param, limit, expression,
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 75, in handler
    raise e
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 67, in handler
    return func(self, *args, **kwargs)
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 104, in handler
    return func(self, *args, **kwargs)
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 628, in search
    return self._execute_search_requests(requests, timeout, **_kwargs)
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 75, in handler
    raise e
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 67, in handler
    return func(self, *args, **kwargs)
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 595, in _execute_search_requests
    raise pre_err
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 586, in _execute_search_requests
    raise BaseException(response.status.error_code, response.status.reason)
pymilvus.client.exceptions.BaseException: <BaseException: (code=1, message=No Available Query node result, filter reason [UnexpectedError] 
: id 429215300261523225)>
 (api_request.py:26)
[2021-11-19 20:55:04,732 - ERROR - ci_test]: (api_response) : <BaseException: (code=1, message=No Available Query node result, filter reason [UnexpectedError] 
: id 429215300261523225)> (api_request.py:27)
FAILED
```

Server 
[milvus_logs.tar.gz](https://github.com/milvus-io/milvus/files/7570390/milvus_logs.tar.gz)
logs:

### Expected Behavior

Always return search results

### Steps To Reproduce

```markdown
Run test case twice, the first time it succeeds and the second time it fails

    def test_compact_after_binary_index(self):
        """
        target: test compact after create index
        method: 1.insert binary data into two segments
                2.create binary index
                3.compact
                4.search
        expected: Verify segment info and index info
        """
        collection_w = self.init_collection_wrap(name=cf.gen_unique_str(prefix), shards_num=1,
                                                 schema=cf.gen_default_binary_collection_schema())

        df, _ = cf.gen_default_binary_dataframe_data(ct.default_nb)
        for i in range(2):
            collection_w.insert(data=df)
            assert collection_w.num_entities == (i+1) * ct.default_nb

        # create index
        collection_w.create_index(ct.default_binary_vec_field_name, ct.default_binary_index)
        log.debug(collection_w.index())

        # compact
        collection_w.compact()
        collection_w.wait_for_compaction_completed()
        collection_w.get_compaction_plans()

        # search
        collection_w.load()
        search_res, _ = collection_w.search(cf.gen_binary_vectors(ct.default_nq, ct.default_dim)[1],
                                            ct.default_binary_vec_field_name,
                                            ct.default_search_params, ct.default_limit)
        assert len(search_res) == ct.default_nq
        for hits in search_res:
            assert len(hits) == ct.default_limit
```
```


### Anything else?

_No response_

**Comment by ThreadDao** [comment]
For similar issues refer to #11406

**Comment by ThreadDao** [comment]
@xiaofan-luan #11406 has been fixed 11 days ago, but I encountered this error again using the image just three days ago (`master-20211119-8a9fb9b`). So please @DragonDriver take a loot at the log first!

**Comment by DragonDriver** [comment]
The reason differs from #11406 . Try to reproduce in my local pc.
/assign

**Comment by DragonDriver** [comment]
Binary vector with `L2` is not supported, try `IP` or `JACCARD` instead.

**Comment by ThreadDao** [comment]
Fixed on image `master-20211123-73f18c5`
